pubkey_encrypt.module

```diff
@@ -180,3 +192,131 @@ function pubkey_encrypt_user_role_update(\Drupal\user\RoleInterface $role) {
     $pubkey_encrypt_manager->updateAllRoleKeys();
   }
 }
+
+/**
+ * Initialize the Pubkey Encrypt module via Batch API.
+ */
+function pubkey_encrypt_initialize_module() {
+  $batch = array(
+    'title' => t('Initializing the Pubkey Encrypt module.'),
+    'operations' => array(),
+    'progress_message' => t('Completed @current out of @total operations.'),
+    'finished' => '_pubkey_encrypt_module_initialize_batch_finished',
+  );
+
+  // Add operations in the batch related to User Keys generation.
+  $users = \Drupal::entityTypeManager()->getStorage('user')->loadMultiple();
+  foreach ($users as $user) {
+    $batch['operations'][] = array(
+      '_pubkey_encrypt_initialize_user_keys',
+      array($user),
+    );
+  }
+
+  // Add operations in the batch related to Role Keys generation and Encryption
+  // Profiles generation.
+  $roles = \Drupal::entityTypeManager()->getStorage('user_role')->loadMultiple();
+  foreach ($roles as $role) {
+    if ($role->id() != UserInterface::ANONYMOUS_ROLE && $role->id() != UserInterface::AUTHENTICATED_ROLE) {
+      $batch['operations'][] = array(
+        '_pubkey_encrypt_initialize_role_key',
+        array($role),
+      );
+      $batch['operations'][] = array(
+        '_pubkey_encrypt_initialize_encryption_profile',
+        array($role),
+      );
+    }
+  }
+
+  batch_set($batch);
+  // Batch processing is handled via the Form API where this function is called.
+}
+
+/**
+ * Implements callback_batch_operation().
+ *
+ * Generate the Asymmetric keys for a user.
+ *
+ * @param \Drupal\user\UserInterface $user
+ *   The user for which keys will be created.
+ */
+function _pubkey_encrypt_initialize_user_keys(UserInterface $user, &$context) {
+  $pubkey_encrypt_manager = \Drupal::service('pubkey_encrypt.pubkey_encrypt_manager');
+  $pubkey_encrypt_manager->initializeUserKeys($user);
+
+  $context['message'] = t('Currently processing operations related to User Keys generation.');
+}
+
+/**
+ * Implements callback_batch_operation().
+ *
+ * Generate the Role key for a role.
+ *
+ * @param \Drupal\user\Entity\Role $role
+ *   The role for which a Role key will be created.
+ */
+function _pubkey_encrypt_initialize_role_key(Role $role, &$context) {
+  $pubkey_encrypt_manager = \Drupal::service('pubkey_encrypt.pubkey_encrypt_manager');
+  $pubkey_encrypt_manager->enableRole($role);
+  $pubkey_encrypt_manager->generateRoleKey($role);
+
+  $context['message'] = t('Currently processing operations related to Role Keys generation.');
```

Comment:
It would be nicer to provide a more specific message like "Initializing role key for %role_name."